File: backend/customer_service/models/customer.py
"""CS Customer ORM model

Maps to ``customer_service.customers`` table (new in Alembic 0003).
"""
from sqlalchemy import (
    Column, BigInteger, String, Text, DateTime, Index,
)
from sqlalchemy.dialects.postgresql import JSONB
from datetime import datetime, timezone

from backend.customer_service.models.conversation import CSBase

_now = lambda: datetime.now(timezone.utc)


class CSCustomer(CSBase):
    __tablename__ = "customers"
    __table_args__ = (
        Index("idx_cs_customer_external", "auth_provider", "auth_external_id", unique=True),
        {"schema": "customer_service"},
    )

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    customer_id = Column(String(64), unique=True, nullable=False)

    display_name = Column(String(128), nullable=False, default="")
    email = Column(String(128), nullable=True)
    phone = Column(String(20), nullable=True)
    avatar_url = Column(Text, nullable=True)

    auth_provider = Column(String(20), nullable=True)
    auth_external_id = Column(String(128), nullable=True)

    custom_fields = Column(JSONB, nullable=False, default=dict)

    created_at = Column(DateTime(timezone=True), default=_now, nullable=False)
    updated_at = Column(DateTime(timezone=True), default=_now, onupdate=_now, nullable=False)

File: business-service/src/main/java/com/agent/cs/domain/Customer.java
package com.agent.cs.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.OffsetDateTime;
import java.util.Map;

/**
 * 客户 — customer_service.customers
 */
@Entity
@Table(name = "customers", schema = "customer_service")
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "customer_id", nullable = false, unique = true, length = 64)
    private String customerId;

    @Column(name = "display_name", nullable = false, length = 128)
    private String displayName = "";

    @Column(name = "email", length = 128)
    private String email;

    @Column(name = "phone", length = 20)
    private String phone;

    @Column(name = "avatar_url", columnDefinition = "text")
    private String avatarUrl;

    @Column(name = "auth_provider", length = 20)
    private String authProvider;

    @Column(name = "auth_external_id", length = 128)
    private String authExternalId;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "custom_fields", columnDefinition = "jsonb", nullable = false)
    private Map<String, Object> customFields = Map.of();

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt = OffsetDateTime.now();

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt = OffsetDateTime.now();

    // ── getters / setters ──

    public Long getId() { return id; }
    public String getCustomerId() { return customerId; }
    public void setCustomerId(String customerId) { this.customerId = customerId; }
    public String getDisplayName() { return displayName; }
    public void setDisplayName(String displayName) { this.displayName = displayName; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }
    public String getAvatarUrl() { return avatarUrl; }
    public void setAvatarUrl(String avatarUrl) { this.avatarUrl = avatarUrl; }
    public String getAuthProvider() { return authProvider; }
    public void setAuthProvider(String authProvider) { this.authProvider = authProvider; }
    public String getAuthExternalId() { return authExternalId; }
    public void setAuthExternalId(String authExternalId) { this.authExternalId = authExternalId; }
    public Map<String, Object> getCustomFields() { return customFields; }
    public void setCustomFields(Map<String, Object> customFields) { this.customFields = customFields; }
    public OffsetDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(OffsetDateTime createdAt) { this.createdAt = createdAt; }
    public OffsetDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(OffsetDateTime updatedAt) { this.updatedAt = updatedAt; }
}
